File: in_Java/0071 Simplify Path.java
class Solution {
    public String simplifyPath(String path) {
        Stack<String> st = new Stack<>();
        int start;
        int end;
        for(int i = 0; i < path.length(); ){
            while(i < path.length() && path.charAt(i) == '/') i++;
            start = i;
            while(i < path.length() && path.charAt(i) != '/') i++;
            end = i;
            i++;
            String cur = path.substring(start,end);//System.out.println(cur);
            if(cur.equals(".") || cur.equals("")) continue;
            else if(cur.equals("..")){
                if(!st.isEmpty()) st.pop();
            }else st.push(cur);
        }
        StringBuilder sb = new StringBuilder();
        if(st.isEmpty()) return "/";
        while(!st.isEmpty()){
            sb.insert(0,st.pop());
            sb.insert(0,"/");
        }
        return sb.toString();
    }
}

File: in_Python/0071 Simplify Path.py
class Solution(object):
    def simplifyPath(self, path):
        """
        :type path: str
        :rtype: str
        """
        s = []
        paths = path.split('/')
        print(paths)
        actualpaths = []
        for p in paths:
            if p == '':
                continue
            elif p == '.':
                continue
            elif p == '..':
                if len(actualpaths) != 0:
                    actualpaths.pop()
            else:
                actualpaths.append(p)
        res = '/'
        for i in actualpaths:
            res = res + i + '/'
        if len(res) <= 1:
            return res
        return res[0 : -1]